samples: dect: dect_phy: dect_shell: new sample shell application for dect_phy (#17627)
nrfconnect/sdk-nrf: samples/dect/dect_phy/dect_shell/README.rst

* Client side - pro tip: decrease default scheduler delay and rerun the previous step. Note: remember to set it back to default to avoid scheduler problems on other use cases.

  .. code-block:: console

     dect sett -d 5000

----

RX/TX testing with RF tool
==========================

DeSh command: ``dect rf_tool``

RF tool is for RX/TX testing. Can be used as a util for running of ETSI EN 301 406-2: sections 4.3 (Conformance requirements for transmitter) and 4.4 (Conformance requirements for receiver).

Frame is sent ``--frame_repeat_count`` times and results are reported. No delay between frames.
Frames up to ``--frame_repeat_count`` defines an interval. Intervals are repeated until ``--frame_repeat_count_intervals``.
There is also a continuous mode that can be started by using ``--continuous`` option.
Continuous mode can be stopped by using ``dect rf_tool stop``.

Frame structure is configured with command options:

  .. code-block:: console

     rx_frame_start_offset : rx_subslot_count + rx_idle_subslot_count + tx_frame_start_offset + tx_subslot_count + tx_idle_subslot_count

> Example #1: bi-directional testing
> ----------------------------------

Reviewer: ```suggestion
Example 1: bi-directional testing
---------------------------------
```

> * On both client and in server side - scan for a free channel:

Reviewer: ```suggestion
* On both client and server side - scan for a free channel:
```

  .. code-block:: console

     dect rssi_scan -c 0

> * "Server" side: Start a RF mode RX_TX with default frame structure and put it waiting for a RX synch:

Reviewer: ```suggestion
* Server side: Start RF mode RX_TX with default frame structure and put it waiting for RX synch:
```

  .. code-block:: console

     dect sett -t 39
     dect rf_tool -m rx_tx --rx_find_sync --frame_repeat_count 50 --frame_repeat_count_intervals 10 -c 1677

> * "Client" side: Trigger to start operation:

Reviewer: ```suggestion
* Client side: Trigger to start operation:
```

  .. code-block:: console

     dect rf_tool -m rx_tx --frame_repeat_count 50 --frame_repeat_count_intervals 10 -t 39 -c 1677

> Example #2: unidirectional testing
> ----------------------------------

Reviewer: ```suggestion
Example 2: unidirectional testing
---------------------------------
```

> * On both TX and RX sides - scan for a free channel. See previous example.

Reviewer: ```suggestion
* On both TX and RX side - scan for a free channel (see previous example).
```

> * RX device option #1: RX single shot mode:

Reviewer: ```suggestion
* RX device option 1: RX single shot mode:
```

  .. code-block:: console

     dect sett -t 39
     dect rf_tool -m rx --rx_find_sync --frame_repeat_count 50 -c 1677

> * RX device option #2: RX device on rx_cont mode:

Reviewer: ```suggestion
* RX device option #2: RX device on rx_cont mode:
```
```suggestion
* RX device option 2: RX device on ``rx_cont`` mode:
```

  .. code-block:: console

     dect sett -t 39
     dect rf_tool -m rx_cont -c 1677

> * RX device option #3: RX device on rx_cont mode with the information of TX side to have interval reporting:

Reviewer: ```suggestion
* RX device option 3: RX device in ``rx_cont`` mode with the information of TX side to have interval reporting:
```

  .. code-block:: console

     dect sett -t 39
     dect rf_tool -m rx_cont --rf_mode_peer tx --frame_repeat_count 50 --rx_find_sync -c 1677

* TX device: trigger to start operation:

  .. code-block:: console

     dect rf_tool -m tx --frame_repeat_count 50 -c 1677 -t 39

> * RX device with option #2: stop continuous RX to give a report:

Reviewer: ```suggestion
* RX device with option 2: stop continuous RX to give a report:
```

  .. code-block:: console

     dect rf_tool stop
     cert command stopping.
     RF tool results at transmitter id 39:
     - RX/TX Duty Cycle percentage: ...

> Example #3: duty cycle (RX+TX) testing
> --------------------------------------

Reviewer: ```suggestion
Example 3: duty cycle (RX+TX) testing
-------------------------------------
```

> * On both TX and RX sides - scan for a free channel. See previous examples.

Reviewer: ```suggestion
* On both TX and RX side - scan for a free channel (see previous examples).
```

* Bi-directional testing:

>   RX/TX Duty Cycle percentage: 73.91%:

Reviewer: ```suggestion
  RX/TX duty cycle percentage 73.91%:
```

  .. code-block:: console

     server:
     dect rf_tool -m rx_tx --rx_find_sync --rx_subslot_count 9 --rx_idle_subslot_count 3 --tx_subslot_count 8 --tx_idle_subslot_count 3 --frame_repeat_count 50 -c 1677

     client:
     dect rf_tool -m rx_tx --rx_subslot_count 9 --rx_idle_subslot_count 3 --tx_subslot_count 8 --tx_idle_subslot_count 3 --frame_repeat_count 50 -c 1677 -t 39

>   RX/TX Duty Cycle percentage: 82.50%:

Reviewer: ```suggestion
  RX/TX duty cycle percentage 82.50%:
```

  .. code-block:: console

     server:
     dect rf_tool -m rx_tx --rx_find_sync --rx_subslot_count 17 --rx_idle_subslot_count 3 --tx_subslot_count 16 --tx_idle_subslot_count 4 --frame_repeat_count 50 -c 1677

     client:
     dect rf_tool -m rx_tx --rx_subslot_count 17 --rx_idle_subslot_count 3 --tx_subslot_count 16 --tx_idle_subslot_count 4 --frame_repeat_count 50 -c 1677 -t 39

> * TX / RX testing on separate devices:

Reviewer: ```suggestion
* TX/RX testing on separate devices:
```

>   TX side: RX/TX Duty Cycle percentage: 84.21%:

Reviewer: ```suggestion
  TX side: RX/TX duty cycle percentage 84.21%:
```

  .. code-block:: console

     server:
     dect rf_tool -m rx_cont -c 1677

     client:
     dect rf_tool -m tx --tx_subslot_count 16 --tx_idle_subslot_count 3 -c 1677 -t 39

>   Or alternatively RX side with the information of TX side to have interval reporting:

Reviewer: ```suggestion
  Alternatively, RX side with the information of TX side to have interval reporting:
```

  .. code-block:: console

     server:
     dect rf_tool -m rx_cont --rf_mode_peer tx --tx_subslot_count 16 --tx_idle_subslot_count 3 --rx_find_sync -c 1677

     client:
     dect rf_tool -m tx --tx_subslot_count 16 --tx_idle_subslot_count 3 -c 1677 -t 39

> Example #4: Bi-directional testing with more data
> -------------------------------------------------

Reviewer: ```suggestion
Example 4: Bi-directional testing with more data
------------------------------------------------
```

> * On both TX and RX sides - scan for a free channel. See previous example.

Reviewer: ```suggestion
* On both TX and RX side - scan for a free channel (see previous example):
```

  .. code-block:: console

     server:
     dect rf_tool -m rx_tx --rx_find_sync --rx_subslot_count 9 --rx_idle_subslot_count 4 --tx_subslot_count 8 --tx_idle_subslot_count 4 --tx_mcs 4 --frame_repeat_count 50 -c 1677

     client:
     dect rf_tool -m rx_tx --rx_subslot_count 9 --rx_idle_subslot_count 4 --tx_subslot_count 8 --tx_idle_subslot_count 4 --tx_mcs 4 --frame_repeat_count 50 -c 1677 --tx_pwr 15 -t 39

----

Building
********

.. |sample path| replace:: :file:`samples/dect/dect_shell`

.. include:: /includes/build_and_run_ns.txt

See :ref:`cmake_options` for instructions on how to provide CMake options, for example to use a configuration overlay.

> Dependencies

Reviewer 2: I would mention dk_buttons_and_leds as it's used by the sample.

============

DeSh command: ``dect ping``

Ping is a tool for testing the reachability of a neighbor host running with DeSh ping server.
You can also use it to measure the round-trip time, demonstrate HARQ and for range testing.

The ``ping`` command uses its own proprietary PDU format and protocol.
At end of a ping session, it prints out the statistics of the session.
On the client side, also server side statistics are printed out if the client side ``PING_RESULT_REQ`` is responded with ``PING_RESULT_RESP`` by the server.

Example 1: basic usage
----------------------

* On both client and server side - scan for a free channel:

  .. code-block:: console

     dect rssi_scan -c 0

* Server side: Set a unique transmission ID and start ping server on manually chosen free channel:

  .. code-block:: console

     dect sett -t 39
     dect ping -s --channel 1677

* Client side: start basic pinging:

  .. code-block:: console

     dect ping -c --s_tx_id 39 --channel 1677

* Client side: max amounf of data in one request with custom parameters:

  .. code-block:: console

     dect ping -c --s_tx_id 39 --c_tx_pwr 7 -i 3 -t 2000 -l 4 --c_tx_mcs 4 --c_count 10 --channel 1677

> * Server side: stop ping server

Reviewer: ```suggestion
* Server side: Stop ping server:
```

  .. code-block:: console

     dect ping stop

>    .. note::
>       You can use this command also on the client side to abort pinging.

Reviewer: ```suggestion
  .. note::
     You can use this command also on the client side to abort pinging.
```

Example 2: automatic TX power tuning
------------------------------------

You can use the ``dect ping`` command line option ``--tx_pwr_ctrl_auto`` to enable automatic TX power tuning.
Both sides indicate their expected RSSI level (``--tx_pwr_ctrl_pdu_rx_exp_rssi_level``) on their RX in the sent ping PDU.
Based on that and other information from RX for the received ping PDU, the actual TX power is tuned if the feature is enabled.

* On both client and server side - scan for a free channel (see previous example).
* Server side: Set a unique transmission ID and start the ping server on a manually chosen free channel:

  .. code-block:: console

     dect sett -t 39
     dect ping -s --tx_pwr_ctrl_auto --tx_pwr_ctrl_pdu_rx_exp_rssi_level -55 --channel 1677

> * Client side: start pinging:

Reviewer: ```suggestion
* Client side: Start pinging:
```

  .. code-block:: console

     dect ping -c --s_tx_id 39 --tx_pwr_ctrl_auto --tx_pwr_ctrl_pdu_rx_exp_rssi_level -55 --channel 1677

> 

Reviewer: ```suggestion
```

* On both client and server side - scan for a free channel (see previous example).
* Server side: Set a unique transmission ID and start the ping server on a manually chosen free channel (see previous examples).
* Client side: start basic pinging and request HARQ feedback from the server:

  .. code-block:: console

     dect ping -c --s_tx_id 39 --channel 1677 -h

Throughput performance testing
==============================

DeSh command: ``dect perf``

Perf is a tool for testing the throughput performance.

The ``perf`` command uses its own proprietary PDU format and protocol.
At end of a session, it prints out the statistics.
On the client side, also server side statistics are printed out if the client side ``PERF_RESULT_REQ`` is responded with ``PERF_RESULT_RESP`` by the server.

Example 1: basic usage
----------------------

* On both client and server side - scan for a free channel:

  .. code-block:: console

     dect rssi_scan -c 0

* Server side: Set a unique transmission ID and start the RX server for 15 seconds on a free channel:

  .. code-block:: console

     dect sett -t 39
     dect perf -s -t 15 --channel 1671

> * Server side: as an alternative to the previous command, start the RX server with continuous RX:

Reviewer: ```suggestion
* Server side: As an alternative to the previous command, start the RX server with continuous RX:
```

  .. code-block:: console

     dect sett -t 39
     dect perf -s -t -1 --channel 1671

> * Client side: send with MCS-4 and length of four slots for 10 seconds:

Reviewer: ```suggestion
* Client side: Send with MCS-4 and length of four slots for 10 seconds:
```

  .. code-block:: console

     dect perf -c --c_gap_subslots 3 --c_tx_mcs 4 --c_slots 4 --s_tx_id 39 -t 10 --channel 1671

> * Server side: stop perf server:

Reviewer: ```suggestion
* Server side: Stop perf server:
```

  .. code-block:: console

     dect perf stop

>    .. note::
>       You can use this command also on the client side to abort TX.

Reviewer: ```suggestion
  .. note::
     You can use this command also on the client side to abort TX.
```

Example 2: HARQ
---------------

* On both client and server side - scan for a free channel (see previous example).
* Server side: Set unique transmission ID and start RX server with HARQ + some other tunings:

  .. code-block:: console

     dect sett -t 39
     dect perf -s -t -1 -h --s_harq_feedback_tx_delay_subslots 2 --s_harq_feedback_tx_rx_delay_subslots 3 --channel 1671

* Client side: Send with MCS-4 and length of 4 slots for 10 seconds:

  .. code-block:: console

     dect perf -c --c_gap_subslots 4 --c_tx_mcs 4 --c_slots 4 --s_tx_id 39 -t 12 --c_harq_feedback_rx_delay_subslots 2 --c_harq_feedback_rx_subslots 3 --c_harq_process_nbr_max 7 -h --channel 1671

> * Client side: Decrease default scheduler delay and rerun the previous step.

Reviewer: ```suggestion
* Client side: Decrease default scheduler delay and rerun the previous step:
```

  .. code-block:: console

     dect sett -d 5000

 .. note::
    Set the delay back to default to avoid scheduler problems on other use cases.

RX/TX testing with RF tool
==========================

DeSh command: ``dect rf_tool``

The ``rf_tool`` is meant for RX/TX testing.
You can use it as a util for running of ETSI EN 301 406-2: sections 4.3 (Conformance requirements for transmitter) and 4.4 (Conformance requirements for receiver).

A frame is sent ``--frame_repeat_count`` times and results are reported, no delay between frames.
Frames up to ``--frame_repeat_count`` define the interval.
Intervals are repeated until ``--frame_repeat_count_intervals``.
There is also a continuous mode that you can start with the ``--continuous`` option.
You can stop the continuous mode by using the ``dect rf_tool stop`` option.

To configure the frame structure, use the following command options:

>   .. code-block:: console
> 
>      rx_frame_start_offset : rx_subslot_count + rx_idle_subslot_count + tx_frame_start_offset + tx_subslot_count + tx_idle_subslot_count

Reviewer: ```suggestion
.. code-block:: console

   rx_frame_start_offset : rx_subslot_count + rx_idle_subslot_count + tx_frame_start_offset + tx_subslot_count + tx_idle_subslot_count
```

  .. code-block:: console

     dect rssi_scan -c 0

* Server side: Start RF mode RX_TX with default frame structure and put it waiting for RX synch:

  .. code-block:: console

     dect sett -t 39
     dect rf_tool -m rx_tx --rx_find_sync --frame_repeat_count 50 --frame_repeat_count_intervals 10 -c 1677

* Client side: Trigger to start operation:

  .. code-block:: console

     dect rf_tool -m rx_tx --frame_repeat_count 50 --frame_repeat_count_intervals 10 -t 39 -c 1677

Example 2: unidirectional testing
---------------------------------

* On both TX and RX side - scan for a free channel (see previous example).
* RX device option 1: RX single shot mode:

  .. code-block:: console

     dect sett -t 39
     dect rf_tool -m rx --rx_find_sync --frame_repeat_count 50 -c 1677

* RX device option 2: RX device on ``rx_cont`` mode:

  .. code-block:: console

     dect sett -t 39
     dect rf_tool -m rx_cont -c 1677

* RX device option 3: RX device in ``rx_cont`` mode with the information of TX side to have interval reporting:

  .. code-block:: console

     dect sett -t 39
     dect rf_tool -m rx_cont --rf_mode_peer tx --frame_repeat_count 50 --rx_find_sync -c 1677

> * TX device: trigger to start operation:

Reviewer: ```suggestion
* TX device: Trigger to start operation:
```

  .. code-block:: console

     dect rf_tool -m tx --frame_repeat_count 50 -c 1677 -t 39

> * RX device with option 2: stop continuous RX to give a report:

Reviewer: ```suggestion
* RX device with option 2: Stop continuous RX to give a report:
```

* TX/RX testing on separate devices:

  TX side: RX/TX duty cycle percentage 84.21%:

  .. code-block:: console

     server:
     dect rf_tool -m rx_cont -c 1677

     client:
     dect rf_tool -m tx --tx_subslot_count 16 --tx_idle_subslot_count 3 -c 1677 -t 39

  Alternatively, RX side with the information of TX side to have interval reporting:

  .. code-block:: console

     server:
     dect rf_tool -m rx_cont --rf_mode_peer tx --tx_subslot_count 16 --tx_idle_subslot_count 3 --rx_find_sync -c 1677

     client:
     dect rf_tool -m tx --tx_subslot_count 16 --tx_idle_subslot_count 3 -c 1677 -t 39

Example 4: Bi-directional testing with more data
------------------------------------------------

* On both TX and RX side - scan for a free channel (see previous example):

  .. code-block:: console

     server:
     dect rf_tool -m rx_tx --rx_find_sync --rx_subslot_count 9 --rx_idle_subslot_count 4 --tx_subslot_count 8 --tx_idle_subslot_count 4 --tx_mcs 4 --frame_repeat_count 50 -c 1677

     client:
     dect rf_tool -m rx_tx --rx_subslot_count 9 --rx_idle_subslot_count 4 --tx_subslot_count 8 --tx_idle_subslot_count 4 --tx_mcs 4 --frame_repeat_count 50 -c 1677 --tx_pwr 15 -t 39

> ----
> 

Reviewer: ```suggestion
```

.. _dect_shell_application:

nRF91x1: DECT NR+ PHY Shell
###########################

.. contents::
   :local:
   :depth: 2

The DECT NR+ physical layer (PHY) Shell (DeSh) sample application demonstrates how to set up a DECT NR+ application with the DECT PHY firmware and enables you to test various modem features.

.. important::

>    The sample showcases the use of the :ref:`nrf_modem_dect_phy` interface of the Modem library.

Author: "Modem library" changed to :ref:`nrfxlib:nrf_modem`

Requirements
************

The sample supports the following development kits, and requires at least two kits:

.. table-from-sample-yaml::

> .. include:: /includes/tfm.txt

Author: Good point. I'll add it to sample.yaml

Overview
********

DeSh enables testing of :ref:`nrf_modem_dect_phy` interface and related modem features.
Hence, this sample is not only code sample, which it also is in many aspects, but also a test application for aforementioned features.

The subsections list the DeSh features and show shell command examples for their usage.

.. note::
   To learn more about using a DeSh command, run the command without any parameters.

Main command structure:

  .. code-block:: console

    dect
      status
      sett
      rssi_scan
      perf
      ping
      rf_tool
      rx

Application settings
====================

> DeSh command: ``dect sett``

Author: Yes, no examples of all commands. I will change the order in listing to match of examples

* Server side: Set unique transmission id and start ping server on manually chosen free channel:

  .. code-block:: console

     dect sett -t 39
     dect ping -s --tx_pwr_ctrl_auto --tx_pwr_ctrl_pdu_rx_exp_rssi_level -55 --channel 1677

* Client side: start pinging:

  .. code-block:: console

     dect ping -c --s_tx_id 39 --tx_pwr_ctrl_auto --tx_pwr_ctrl_pdu_rx_exp_rssi_level -55 --channel 1677

Example #3: HARQ
----------------

``dect ping`` command hook ``-h`` can be used to enable client side to request a HARQ feedback for the sent ping requests, and based on received HARQ ACK/NACK, ping request is re-sent if possible within a timeout.

* On both client and in server side - scan for a free channel. See previous example.
* Server side: Set unique transmission id and start ping server on manually chosen free channel - see previous examples.
* Client side: start basic pinging and request HARQ feedback from server:

  .. code-block:: console

     dect ping -c --s_tx_id 39 --channel 1677 -h

----

Throughput performance testing
==============================

DeSh command: ``dect perf``

Perf is a tool for testing the throughput performance.

> Perf command uses its own proprietary PDU format and protocol. At end of a perf session, it prints out the statistics of the session.
> In client side, also server side statistics are printed out if client side PERF_RESULT_REQ is responded with PERF_RESULT_RESP by the server.

Reviewer: ```suggestion
The ``perf`` command uses its own proprietary PDU format and protocol.
At end of a session, it prints out the statistics.
On the client side, also server side statistics are printed out if the client side ``PERF_RESULT_REQ`` is responded with ``PERF_RESULT_RESP`` by the server.
```

Example #1: basic usage
-----------------------

* On both client and in server side - scan for a free channel:

  .. code-block:: console

     dect rssi_scan -c 0

* Server side: Set unique transmission id and start RX server for a 30 seconds in a free channel:

  .. code-block:: console

     dect sett -t 39
     dect perf -s -t 15 --channel 1671

> * Server side: alternatively to previous bullet, start RX server with continuous RX:

Reviewer: ```suggestion
* Server side: as an alternative to the previous command, start the RX server with continuous RX:
```

  .. code-block:: console

     dect sett -t 39
     dect perf -s -t -1 --channel 1671

* Client side: Send with MCS-4 and length of 4 slots for a 10 seconds:

  .. code-block:: console

     dect perf -c --c_gap_subslots 3 --c_tx_mcs 4 --c_slots 4 --s_tx_id 39 -t 10 --channel 1671

> * Server side: stop perf server (note: can be used also in client side to abort client side TX):

Reviewer: ```suggestion
* Server side: stop perf server:
```

  .. code-block:: console

     dect perf -c --c_gap_subslots 4 --c_tx_mcs 4 --c_slots 4 --s_tx_id 39 -t 12 --c_harq_feedback_rx_delay_subslots 2 --c_harq_feedback_rx_subslots 3 --c_harq_process_nbr_max 7 -h --channel 1671

* Client side - pro tip: decrease default scheduler delay and rerun the previous step. Note: remember to set it back to default to avoid scheduler problems on other use cases.

  .. code-block:: console

     dect sett -d 5000

----

RX/TX testing with RF tool
==========================

DeSh command: ``dect rf_tool``

RF tool is for RX/TX testing. Can be used as a util for running of ETSI EN 301 406-2: sections 4.3 (Conformance requirements for transmitter) and 4.4 (Conformance requirements for receiver).

Frame is sent ``--frame_repeat_count`` times and results are reported. No delay between frames.
Frames up to ``--frame_repeat_count`` defines an interval. Intervals are repeated until ``--frame_repeat_count_intervals``.
There is also a continuous mode that can be started by using ``--continuous`` option.
Continuous mode can be stopped by using ``dect rf_tool stop``.

Frame structure is configured with command options:

  .. code-block:: console

>      rx_frame_start_offset : rx_subslot_count + rx_idle_subslot_count + tx_frame_start_offset + tx_subslot_count + tx_idle_subslot_count

Author: Trying to explain here the frame structure based on different command hooks.

Example #1: bi-directional testing
----------------------------------

* On both client and in server side - scan for a free channel:

  .. code-block:: console

     dect rssi_scan -c 0

* "Server" side: Start a RF mode RX_TX with default frame structure and put it waiting for a RX synch:

  .. code-block:: console

     dect sett -t 39
     dect rf_tool -m rx_tx --rx_find_sync --frame_repeat_count 50 --frame_repeat_count_intervals 10 -c 1677

* "Client" side: Trigger to start operation:

  .. code-block:: console

     dect rf_tool -m rx_tx --frame_repeat_count 50 --frame_repeat_count_intervals 10 -t 39 -c 1677

> Example #2: TX / RX testing on separate devices

Author: Yes, unidirectional. That is said very complex, that TX and RX parts are in separate devices :-) . 
Changed this as:
"Example #2: unidirectional testing"

-----------------------------------------------

* On both TX and RX sides - scan for a free channel. See previous example.
* RX device option #1: RX single shot mode:

  .. code-block:: console

     dect sett -t 39
     dect rf_tool -m rx --rx_find_sync --frame_repeat_count 50 -c 1677

* RX device option #2: RX device on rx_cont mode:

  .. code-block:: console

     dect sett -t 39
     dect rf_tool -m rx_cont -c 1677

* RX device option #3: RX device on rx_cont mode with the information of TX side to have interval reporting:

  .. code-block:: console

     dect sett -t 39
     dect rf_tool -m rx_cont --rf_mode_peer tx --frame_repeat_count 50 --rx_find_sync -c 1677

* TX device: trigger to start operation:

  .. code-block:: console

     dect rf_tool -m tx --frame_repeat_count 50 -c 1677 -t 39

* RX device with option #2: stop continuous RX to give a report:

  .. code-block:: console

     dect rf_tool stop
     cert command stopping.
     RF tool results at transmitter id 39:
     - RX/TX Duty Cycle percentage: ...

> Example #3: Higher duty cycle (RX+TX) hunting

Author: Rephrased to: Example #3: duty cycle (RX+TX) testing

  Or alternatively RX side with the information of TX side to have interval reporting:

  .. code-block:: console

     server:
     dect rf_tool -m rx_cont --rf_mode_peer tx --tx_subslot_count 16 --tx_idle_subslot_count 3 --rx_find_sync -c 1677

     client:
     dect rf_tool -m tx --tx_subslot_count 16 --tx_idle_subslot_count 3 -c 1677 -t 39

Example #4: Bi-directional testing with more data
-------------------------------------------------

* On both TX and RX sides - scan for a free channel. See previous example.

  .. code-block:: console

     server:
     dect rf_tool -m rx_tx --rx_find_sync --rx_subslot_count 9 --rx_idle_subslot_count 4 --tx_subslot_count 8 --tx_idle_subslot_count 4 --tx_mcs 4 --frame_repeat_count 50 -c 1677

     client:
     dect rf_tool -m rx_tx --rx_subslot_count 9 --rx_idle_subslot_count 4 --tx_subslot_count 8 --tx_idle_subslot_count 4 --tx_mcs 4 --frame_repeat_count 50 -c 1677 --tx_pwr 15 -t 39

----

Configuration
*************

|config|

Configuration options
=====================

Check and configure the following configuration options for the sample:

.. _CONFIG_DESH_DECT_PHY:

> CONFIG_DESH_DECT_PHY

Author: It is not meaningful to be mentioned here currently.

> CONFIG_DESH_DECT_PHY

Author: Removed "Configuration" chapter
